src/simple_experiments/events/simple_card.cljs

```diff
@@ -25,23 +19,63 @@
     :pending-registration
     :associated})
 
-(defn set-active-card [{:keys [db]} [_ card-uuid status]]
-  (if (nil? card-uuid)
-    {}
-    (let [sdid (six-digit-id card-uuid)]
-      {:db (assoc-in db [:ui :active-card]
-                     {:uuid card-uuid
-                      :six-digit-id sdid
-                      :six-digit-display (six-digit-display sdid)
-                      :status status})})))
+(defn has-six-digit-id? [patient six-digit-id]
+  (let [six-digit-ids (set (concat (:six-digit-ids patient)
+                                   (map ->six-digit-id
+                                        (:card-uuids patient))))]
+    (contains? six-digit-ids six-digit-id)))
```

Comment:
We wouldn't need to check in both uuids and six-digit-ids. Ideally, the six-digit-ids is just a cached conversion of UUIDs, and potentially some unmapped ones. But we would need to look up only in six-digit-ids. Not for performance reasons, but for simplicity.

Reply:
Intertwining six digits and uuids would make it more complex, not simpler